src/Lecture1.hs

```diff
@@ -94,7 +101,15 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str = drop startix (take (endix + 1) str) where
+    startix
+        | start >= 0 = start
+        | otherwise  = 0
```

Comment:
You can write this function shorter by using standard function `max`:

```suggestion
    startix = max 0 start
```

The same is true for `endix`